.github/workflows/main.yml

```diff
@@ -25,40 +25,56 @@ jobs:
     steps:
     - uses: actions/checkout@v2
 
+    # Java 11 is specifically needed for Checkstyle.
+    - name: Set up JDK 1.11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.11
+
+    - name: Download Checkstyle
```

Comment:
Do we still need this if the script downloads it?

Reply:
yes, right now, I cannot use our setup script here. 
Setup script is doing two things
1. copying the pre-push script to hooks
2. download all the required setup files for ktlint etc

Reason 1- point 1 is not needed in the GitHub action.
Reason 2- GitHub action is not understanding the `../oppia-android-tools`, so I had mentioned `$HOME` for now. Creating a separate issue for this where we can have a balanced use of pre-push scripts and GitHub action altogether.

Reply:
Ideally we should try to fix that in this PR for the Java linter rather than checking in two places that both have separate linter versions.

I'm actually not suggesting that we run the setup script. Instead, could we move out the dependencies bits into a "setup_deps.sh" script & use that in actions, instead? Then "setup.sh" would call "setup_deps.sh" & set up the pre-push hook.

Reply:
Done, thanks for the suggestion.